File: WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/ImageBlockProcessor.java
package org.wordpress.android.ui.posts.mediauploadcompletionprocessors;

import androidx.annotation.NonNull;

import com.google.gson.JsonElement;
import com.google.gson.JsonObject;

import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.wordpress.android.util.helpers.MediaFile;


public class ImageBlockProcessor extends BlockProcessor {
    public ImageBlockProcessor(@NonNull String localId, @NonNull MediaFile mediaFile) {
        super(localId, mediaFile);
    }

    @Override
    public boolean processBlockContentDocument(@NonNull Document document) {
        // select image element with our local id
        Element targetImg = document.select("img").first();

        // if a match is found, proceed with replacement
        if (targetImg != null) {
            // replace attributes
            targetImg.attr("src", remoteUrl);

            // replace class
            targetImg.removeClass("wp-image-" + localId);
            targetImg.addClass("wp-image-" + remoteId);

            return true;
        }

        return false;
    }

    @Override
    public boolean processBlockJsonAttributes(@NonNull JsonObject jsonAttributes) {
        JsonElement id = jsonAttributes.get("id");
        if (id != null && !id.isJsonNull() && id.getAsString().equals(localId)) {
            addIntPropertySafely(jsonAttributes, "id", remoteId);
            return true;
        }
        return false;
    }
}

File: WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/ImageBlockProcessor.kt
package org.wordpress.android.ui.posts.mediauploadcompletionprocessors

import com.google.gson.JsonObject
import org.jsoup.nodes.Document
import org.wordpress.android.util.helpers.MediaFile

class ImageBlockProcessor(localId: String, mediaFile: MediaFile) : BlockProcessor(localId, mediaFile) {
    override fun processBlockContentDocument(document: Document): Boolean {
        // select image element with our local id
        val targetImg = document.select("img").first()

        // if a match is found, proceed with replacement
        return targetImg?.let {
            // replace attributes
            it.attr("src", remoteUrl)

            // replace class
            it.removeClass("wp-image-$localId")
            it.addClass("wp-image-$remoteId")

            true
        } ?: false
    }

    override fun processBlockJsonAttributes(jsonAttributes: JsonObject): Boolean {
        val id = jsonAttributes["id"]
        return if (id != null && !id.isJsonNull && id.asString == localId) {
            addIntPropertySafely(jsonAttributes, "id", remoteId)
            true
        } else {
            false
        }
    }
}
